baseframe/forms/sqlalchemy.py

```diff
@@ -25,6 +25,8 @@ def __init__(self, attr, message=None, model=None):
 
     def __call__(self, form, field):
         model = self.model or form.edit_model
+        if not model:
+            raise TypeError(u"Either the field or the form MUST be linked to a model")
```

Comment:
`self` is the validator, not the field. Either the validator or the form must be linked to a model.